pkg/crd/operator/v1alpha1/configuration.go

```diff
@@ -20,13 +20,34 @@ import (
 	"github.com/Masterminds/semver/v3"
 
 	kubermaticv1 "k8c.io/kubermatic/v2/pkg/crd/kubermatic/v1"
-	"k8c.io/kubermatic/v2/pkg/version"
 
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/sets"
 )
 
+// ConditionType is the type defining the cluster or datacenter condition that must be met to block a specific version
+type ConditionType string
+
+const (
+	// AlwaysCondition represent an always true matching condition used while checking provider incompatibilities
+	AlwaysCondition ConditionType = "always"
+	// ExternalCloudProviderCondition is an incompatibility condition that represents the usage of the external Cloud Provider
+	ExternalCloudProviderCondition ConditionType = kubermaticv1.ClusterFeatureExternalCloudProvider
+)
+
+// OperationType is the type defining the operations triggering the compatibility check (CREATE or UPDATE)
+type OperationType string
+
+const (
+	// CreateOperation represents the creation of a new cluster
+	CreateOperation OperationType = "CREATE"
+	// UpdateOperation represents the update of an existing cluster
+	UpdateOperation OperationType = "UPGRADE"
```

Comment:
NIT: should it be UPDATE? or the var should be called UpgradeOperation and the description updated

Reply:
We have many functions called "controlPlaneUpgrade", so we have 2 distinct names pretty much everywhere for this. I agree that it's ugly, but it's not this PR's job to clean that up ^^